RingPublishingGDPR/src/main/java/com/ringpublishing/gdpr/internal/model/TenantConfiguration.java

```diff
@@ -0,0 +1,47 @@
+package com.ringpublishing.gdpr.internal.model;
+
+public class TenantConfiguration
+{
+    public enum TenantState {
```

Comment:
dziwne że konfiguracja ma stan a nie po prostu jest albo jej nie ma :)

Reply:
Jak jest nulem lub nie, to wiesz tylko że jest. Nie wiesz czy na nią czekać. Czy nadal jest w trakcie pobierania lub wystapił błąd przy jej pobraniu. 

Można ewentualnie wydzielić ten stan poza konfigurację ale tu jest najbliżej obiektu, którego dotyczy.